### ISSUE TITLE
Search bar

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#4149]
# Search bar

## Brief description

This component is to be migrated from the `@carbon/ibm-security` package (found in `main_v1` packages/security) to our `main` branch and is part of the initiative documented in #4052 and #4036.

Migrate `SearchBar` from Products v1 / Security / TruncatedList to Products v2.

References: [GitHub](https://github.com/carbon-design-system/ibm-products/blob/main_v1/packages/security/src/components/SearchBar/SearchBar.js), [Storybook](https://v1-ibm-products.carbondesignsystem.com/?path=/story/security-patterns-searchbar--default).

## Details

[Link to Security version](https://v1-ibm-products.carbondesignsystem.com/?path=/story/security-patterns-searchbar--default)

**Notes on this migration**
For this initial release, we will follow these two principles to keep working moving forward and to unblock teams reliant on the older package —
1. **Zero design changes from Security to Products.** This make it easier to compare old component to new component. Any design updates can as followup.
2. **New components must meet Products’ code specs.** Rather than copy and paste old code from Security to Products, we will refactor it to ensure they’re up to date.

Summary from @paul-balchin-ibm’s initial review —

> ### SearchBar
>
>[GitHub](https://github.com/carbon-design-system/ibm-products/blob/main_v1/packages/security/src/components/SearchBar/SearchBar.js), [Storybook](https://v1-ibm-products.carbondesignsystem.com/?path=/story/security-patterns-searchbar--default)
>
>#### TLDR
>
> - **✗** See `This component must be refactored from a Class to a Function`. ([Link](https://github.com/carbon-design-system/ibm-products/issues/4052))
>
> #### PROBLEMATIC IMPORTS
>
> ```
> import Button from '../Button';
> import Search from '../Search';
> import MultiSelect from '../MultiSelect';
> ```

### Links to other materials

- The level of work required for all components

### Design contacts
- @sdignum

## Tasks

Before starting work on this epic, please review and complete the following.

- [x] Read the [Overview of the Security migration](https://github.com/carbon-design-system/ibm-products/issues/4052)
- [x] Initial review of design/existing code

### Working in Carbon for IBM Products package

- [x] Have you reviewed our
      [contribution guidelines](https://github.com/carbon-design-system/ibm-products/blob/master/.github/CONTRIBUTING.md)?
- [x] Have you noted our
      [code guidelines](https://github.com/carbon-design-system/ibm-products/blob/master/docs/CODE_GUIDELINES.md)?
- [x] Did you use the CLI to scaffold your new component?

```[tasklist]
### Acceptance criteria
- [x] Migrate component to `main` under IBM Products
- [ ] https://github.com/carbon-design-system/ibm-products/issues/4239
```
